internal/datastreams/processor.go

```diff
@@ -468,8 +472,10 @@ func (p *Processor) SetCheckpointWithParams(ctx context.Context, params options.
 	if params.ServiceOverride != "" {
 		service = params.ServiceOverride
 	}
+	processTags := processtags.GlobalTags().Slice()
 	child := Pathway{
-		hash:         p.hashCache.get(service, p.env, edgeTags, parentHash),
+		// FIXME: possible inconsistency, processtags are read here but also in the flush method, values could differ if someone calls processtags.Add in the middle.
```

Comment:
that's not an issue as I see it. If process tags are changed in the middle of execution, there will be some inconsistencies somewhere anyway.